### ISSUE TITLE
FieldProxy improvement

### ISSUE THREAD

**Issue by bric3** [raphw/byte-buddy#158]
Hi @raphw 

I recently tried the `FieldProxy` class, and looking at the Javadoc only I thought there could be some improvements, currently the javadoc says 

```
The getter type must be defined in a single-method interface
with a single method that returns an {@link java.lang.Object} type and takes no arguments. The getter interface
must similarly return {@code void} and take a single {@link java.lang.Object} argument.
```

The javadoc could advise to use generics this way : 

```java
public interface Get<T> {
    T get();
}

public interface Set<T> {
    void set(T value);
}
```

Also this line `Using this annotation it is possible to access fields by getter and setter types.` could be strengthen the point that is either a getter **or** a setter type, if one wants both to read and set a field he should declare two parameters.

```java
@FieldProxy("buffer") Get<ByteBuffer> bufferGetter,
@FieldProxy("buffer") Set<ByteBuffer> bufferSetter,
```

Also why not allowing this kind of interface (typed and two methods) ?

```java
public interface BufferAccessor {
    ByteBuffer get();
    void set(ByteBuffer value);
}
```

**Comment by raphw** [comment]
You are right, the `FieldProxy` can be much better. I will add this improvement in a future version.

**Comment by raphw** [comment]
I looked at this yesterday night and remembered why I split this up. Declaring a setter proxy exlcudes an interceptor if the located field is `final`. This allows for easy resolution. I did however add an overloaded version of the installer where a combined getter/setter proxy type can be specified. If the setter is called on a `final` field, this proxy now throws an `UnsupportedOperationException`. A user can now choose between the two depending on which one is more appropriate.

Fixed on head.
